src/main/java/Duke.java

```diff
@@ -1,10 +1,147 @@
+                    listNum++;
+                }
+            }
+            break;
+        case COMMAND_MARK:
+            int markNum = Integer.parseInt(commandArr[1]) - 1;
+            taskItems[markNum].isDone = true;
+            System.out.println("Nice! I've marked this task as done:\n" + "[" + taskItems[markNum].taskType + "]"+ "[X] " + taskItems[markNum].description);
+            break;
+        case COMMAND_UNMARK:
+            int unMarkNum = Integer.parseInt(commandArr[1]) - 1;
+            taskItems[unMarkNum].isDone = false;
+            System.out.println("Nice! I've marked this task as not done yet:\n" + "[" + taskItems[unMarkNum].taskType + "]" + "[ ] " + taskItems[unMarkNum].description);
+            break;
+        case COMMAND_BYE:
+            exitProgram();
+            break;
+        default:
+            addTask(inCommand);
+            break;
+        }
+    }
+
+    private static void exitProgram() {
+        System.out.println("Bye. Hope to see you again soon!");
+        System.exit(0);
+    }
+
+    private static void addTask(String inCommand) {
+        String[] sentenceArr = inCommand.split(" ");
+        String[] taskToInput;
+        String printTask;
+        switch (sentenceArr[0]) {
+        case COMMAND_TODO:
+            taskToInput = Arrays.copyOfRange(sentenceArr, 1, sentenceArr.length);
+            printTask = arrayToString(taskToInput);
+            taskItems[itemNum] = new Task(printTask);
+            taskItems[itemNum].taskType = 'T';
+            System.out.println("Got it. I've added this task:\n" + "[T][ ] " + printTask +
+                    "\nNow you have " + (itemNum + 1) +  " tasks in the list.");
+            itemNum++;
+            break;
+        case COMMAND_DEADLINE:
+            int getDateIndex = Arrays.asList(sentenceArr).indexOf("/by");
+            taskToInput = Arrays.copyOfRange(sentenceArr, 1, getDateIndex);
+            String[] date = Arrays.copyOfRange(sentenceArr, getDateIndex + 1, sentenceArr.length);
+            String[] finalDeadlineArr = concatArray(taskToInput, date, " (by:");
+            printTask = arrayToString(finalDeadlineArr);
+            taskItems[itemNum] = new Task(printTask);
+            taskItems[itemNum].taskType = 'D';
+            System.out.println("Got it. I've added this task:\n" + "[D][ ] " + printTask +
+                    "\nNow you have " + (itemNum + 1) +  " tasks in the list.");
+            itemNum++;
+            break;
+        case COMMAND_EVENT:
+            int getTimeIndex = Arrays.asList(sentenceArr).indexOf("/at");
+            taskToInput = Arrays.copyOfRange(sentenceArr, 1, getTimeIndex);
+            String[] timing = Arrays.copyOfRange(sentenceArr, getTimeIndex + 1, sentenceArr.length);
+            String[] finalEventArr = concatArray(taskToInput, timing, "(at:");
```

Comment:
Use an Event.java class instead? I think this was mentioned/suggested in the question of this assignment